tests/publiccloud/ipa.pm

```diff
@@ -0,0 +1,162 @@
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+use File::Basename;
+use serial_terminal 'select_virtio_console';
+
+
+sub find_secgroup {
+    my ($name) = @_;
+
+    my $out = script_output("aws ec2 describe-security-groups --group-names '$name'", 30, proceed_on_failure => 1);
+    if ($out =~ /"GroupId":\s+"([^"]+)"/) {
+        return $1;
+    }
+    return;
+}
+
+sub save_logs {
+    my $output = script_output("find ipa_results -type f");
+    for my $file (split(/\n/, $output)) {
+        if ($file =~ m'ipa_results/ec2/ami-[0-9a-z]+/i-[0-9a-z]+/[0-9]{14}\.(log|results)') {
+            upload_logs($file);
+        }
+    }
+}
+
+sub run {
+
+    select_virtio_console();
+
+    die "Public cloud provider isn't supported" unless check_var('PUBLIC_CLOUD_PROVIDER', "EC2");
+
+    zypper_call('ar ' . get_required_var('PUBLIC_CLOUD_TOOLS_REPO'));
+    zypper_call('--gpg-auto-import-keys in python3-ipa python3-ipa-tests');
+
+    # WAR install awscli from pip instead of using the package bcs#1095041
+    zypper_call('in gcc python3-pip');
+    if (check_var('DISTRI', 'opensuse')) {
+        zypper_call('in python3-devel');
+        assert_script_run("pip3 install pycrypto");
+    }
+    assert_script_run("pip3 install awscli");
+    assert_script_run("pip3 install keyring");
+
+    assert_script_run("export AWS_ACCESS_KEY_ID=" . get_required_var('PUBLIC_CLOUD_KEY_ID'));
+    assert_script_run("export AWS_SECRET_ACCESS_KEY=" . get_required_var('PUBLIC_CLOUD_KEY_SECRET'));
+    assert_script_run("export AWS_DEFAULT_REGION='eu-central-1'");
+
+    # Create SSH key
+    assert_script_run("aws ec2 delete-key-pair --key-name QA_SSH_KEY");
+    assert_script_run("aws ec2 create-key-pair --key-name QA_SSH_KEY --query 'KeyMaterial' --output text > QA_SSH_KEY.pem");
+
+    # Create/find security group
+    my $secgroup_id = find_secgroup("qa_secgroup");
+    if (!defined($secgroup_id)) {
+        assert_script_run("aws ec2 create-security-group --group-name qa_secgroup --description 'SSH_OPEN'");
+        assert_script_run("aws ec2 authorize-security-group-ingress --group-name qa_secgroup --protocol tcp --port 22 --cidr 0.0.0.0/0");
+        assert_script_run("aws ec2 authorize-security-group-ingress --group-name qa_secgroup --protocol icmp --cidr 0.0.0.0/0 --port 0 ");
+        $secgroup_id = find_secgroup("qa_secgroup");
+        die "Failed on creating security group " unless $secgroup_id;
+    }
+
+    my $ami_id = get_required_var("PUBLIC_CLOUD_IMAGE_ID");
+
+    #Prestart instance, cause IPA might use the wrong security group
+    my ($instance_id)
+      = script_output("aws ec2 run-instances --image-id $ami_id --instance-type t2.large --key-name QA_SSH_KEY --security-group-ids qa_secgroup")
+      =~ /"InstanceId":\s*"([^"]+)"/;
+
+    # download latest IPA tests
+    #    assert_script_run("git clone -q --depth 1 https://github.com/SUSE/ipa.git ipa_repo");
```

Comment:
commented code is bad practice. can you please delete it from PR? If you need it because "you waiting for something  " please keep it in your local branch